### PR TITLE
Feature/update caqti

### PR DESCRIPTION
**Pull request by mabiede**
@joseferben As previously discussed with @aronerben, check the following PR about replacing the `Caqti_request` functions with its infixes.

Additionally, I updated the devcontainer environment to our latest version.

### CODE REVIEW

**Comment by mabiede** [comment]
Nice tests, as described above, the inconsistent parameter style got rejected. I extended the generator to run with non linear and linear substitutions.

File: dune-project [hunk 1]
```diff
@@ -1,132 +1,250 @@
 (lang dune 2.7)
```
Comment (joseferben) [inline dune-project:1]: @mabiede 

The `dune-project` file seems to be formatted weirdly. When I run `make format` with the checked in `.ocamlformat` I get  the formatting before your change. (ocamlformat 0.20.1)

**Comment by joseferben** [comment]
Thanks for the PR! I have one remark regarding the formatting, the caqti change looks good. Thanks for specifying the docker image versions.

@aronerben Do you approve as well?

File: dune-project (hunk 1 above)
Comment (aronerben) [reply dune-project:1]: test

**Comment by aronerben** [comment]
@joseferben @mabiede Yes, looks good to me too, besides the formatting. We should take a look at that.

File: dune-project (hunk 1 above)
Comment (joseferben) [reply dune-project:1]: Thanks for the command, it seems that the VSCode extension is using dune to format dune and dune-project files.

**Pull request merged**